MdeModulePkg/Bus/Pci/NonDiscoverablePciDeviceDxe/NonDiscoverablePciDeviceDxe.c

```
@@ -187,7 +195,42 @@ NonDiscoverablePciDeviceStart (
     goto CloseProtocol;
   }
 
-  Dev->UniqueId = mUniqueIdCounter++;
+  // MU_CHANGE [BEGIN]
+  Status = gBS->HandleProtocol (DeviceHandle, &gEdkiiNonDiscoverableDeviceUniqueIdProtocolGuid, (VOID **)&UniqueIdProtocol);
+  if (Status == EFI_SUCCESS) {
+    if (UniqueIdProtocol->Revision != NON_DISCOVERABLE_DEVICE_UNIQUE_ID_PROTOCOL_REVISION) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "%a: UniqueId protocol revision mismatch. Expected %d, got %d\n",
+        __func__,
+        NON_DISCOVERABLE_DEVICE_UNIQUE_ID_PROTOCOL_REVISION,
+        UniqueIdProtocol->Revision
+        ));
+      return EFI_INCOMPATIBLE_VERSION;

```

In these two failure cases, should we assert but still fall back to the static counter? I'm not sure the security implications here.

---

My thought process was that if a platform wanted to publish this protocol, then that would mean their intent is to actually assign a deterministic UniqueId VS the original static counter. If so, it may be better to fail explicitly so that it is clear that they either got the Revision wrong or their UnqueId is not in the allowable range.